packer/vhd-rg-cleanup.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+####################################################
+
+if [ -z "$SERVICE_PRINCIPAL_CLIENT_ID" ]; then
+    echo "must provide a SERVICE_PRINCIPAL_CLIENT_ID env var"
+    exit 1;
+fi
+
+if [ -z "$SERVICE_PRINCIPAL_CLIENT_SECRET" ]; then
+    echo "must provide a SERVICE_PRINCIPAL_CLIENT_SECRET env var"
+    exit 1;
+fi
+
+if [ -z "$TENANT_ID" ]; then
+    echo "must provide a TENANT_ID env var"
+    exit 1;
+fi
+
+if [ -z "$SUBSCRIPTION_ID_TO_CLEANUP" ]; then
+    echo "must provide a SUBSCRIPTION_ID_TO_CLEANUP env var"
+    exit 1;
+fi
+
+if [ -z "$EXPIRATION_IN_HOURS" ]; then
+    EXPIRATION_IN_HOURS=2
+fi
+
+if [ -z "$DRY_RUN" ]; then
+    echo "no DRY_RUN env var provided, will default to true"
+    DRY_RUN=true
+fi
+
+az login --service-principal \
+		--username "${SERVICE_PRINCIPAL_CLIENT_ID}" \
+		--password "${SERVICE_PRINCIPAL_CLIENT_SECRET}" \
+		--tenant "${TENANT_ID}" &>/dev/null
+
+# set to the sub id we want to cleanup
+az account set -s $SUBSCRIPTION_ID_TO_CLEANUP
+
+# convert to seconds so we can compare it against the "tags.now" property in the resource group metadata
+(( expirationInSecs = ${EXPIRATION_IN_HOURS} * 60 * 60 ))
+# deadline = the "date +%s" representation of the oldest age we're willing to keep
+(( deadline=$(date +%s)-${expirationInSecs%.*} ))
+
+# clean up Packer resource groups
+
+# find packer resource groups created before our deadline
+echo "Looking for resource groups created over ${EXPIRATION_IN_HOURS} hours ago..."
+for resourceGroup in $( az group list --query "[?contains(name, 'packer-Resource-Group')]" | jq --arg dl $deadline '.[] | select(.tags.now < $dl).name' | tr -d '\"' || ""); do
```

Comment:
@mboersma do you know any jmespath magic to check whether the value of the now tag is smaller than the deadline var? I tried for a while but couldn't get it to work...

```
 az storage account list -g aksimages --query "[?to_number(tags.now) < $dl]"
az storage account list: error: argument --query: invalid jmespath_type value: '[?to_number(tags.now) < 4]'
```